modules/light-clients/06-solomachine/types/update.go

```diff
@@ -112,6 +123,49 @@ func (cs ClientState) UpdateState(ctx sdk.Context, cdc codec.BinaryCodec, client
 
 	cs.Sequence++
 	cs.ConsensusState = consensusState
+
+	clientStore.Set(host.ClientStateKey(), types.MustMarshalClientState(cdc, &cs))
+
+	// set default consensus height with header height
+	consensusHeight := smHeader.GetHeight()
+	if cs.ClientType() != exported.Localhost {
+		clientStore.Set(host.ConsensusStateKey(consensusHeight), types.MustMarshalConsensusState(cdc, consensusState))
+	} else {
+		consensusHeight = types.GetSelfHeight(ctx)
```

Comment:
where is this coming from? Solo machines don't store consensus states

Reply:
Slightly confused here, isn't [this code](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/keeper/client.go#L100-L106) run for solomachines too?

Reply:
It is, but it shouldn't be. This was another justification for why setting the consensus state within the light client implementation is useful. I thought I documented this clearly, but this was all I could find:

>  This would require light client developers to set the consensus states themselves (allowing header.GetHeight to be removed and stopping unnecessary state storage for solo machines). 

[ref](https://github.com/cosmos/ibc-go/issues/284#issuecomment-988767293)

Solo machines shouldn't be setting a consensus state ever. It has a single consensus state which is constantly updated and thus it is stored within the client state. The code linked above was an inefficiency, performing unnecessary storage

Reply:
Ahh, okay! Tbh I was confused why `ConsensusState` existed as a field on the solomachine `ClientState` and the extra storage was happening.
This should all be much easier to reason about when the refactor is done. Thank you for clarifying this and providing context, much appreciated!

Updated and removed this code :)